readme.md

```diff
@@ -63,6 +63,7 @@ Click the type names for complete docs.
 - [`Except`](source/except.d.ts) - Create a type from an object type without certain keys. This is a stricter version of [`Omit`](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-5.html#the-omit-helper-type).
 - [`Mutable`](source/mutable.d.ts) - Convert an object with `readonly` keys into a mutable object. The inverse of `Readonly<T>`.
 - [`Merge`](source/merge.d.ts) - Merge two types into a new type. Keys of the second type overrides keys of the first type.
+- [`MergeDeep`](source/merge-deep.d.ts) - Merge two types deeply into a new type. Keys of the second type overrides keys of the first type.
```

Comment:
Should we have a runtime reference implementation we're basing it on? Are we basically trying to type `lodash.merge` or is there another?

Reply:
Not sure what you mean here. You'd like to link to `lodash.merge` in the description?

Reply:
Yeah maybe a link would make sense. I've always found it difficult to reason about deep merges, they're complex. e.g. what happens to arrays? What about arrays with complex objects? Having a well established "target" might help make some of those decisions, and give a hint to users how it works.

Reply:
Oh, gotcha. I basically tried to replicate our other `-Deep` types in how this is reasoned about. Just go as deep as possible and when it longer makes sense to go deeper, apply right-hand side type. Not based on lodash, but that might be a mistake I suppose.

Reply:
I don't think we should mention `lodahs.merge` or base it on it, but in (not here) `merge-deep.d.ts`, we should clearly document how the merging works regarding arrays/maps/sets/etc.